src/api/environment.cc

```diff
@@ -67,17 +67,20 @@ MaybeLocal<Value> PrepareStackTraceCallback(Local<Context> context,
   if (env == nullptr) {
     return exception->ToString(context).FromMaybe(Local<Value>());
   }
-  // TODO(legendecas): Per-realm prepareStackTrace callback.
-  // If we are in a Realm that is not the principal Realm (e.g. ShadowRealm),
-  // skip the prepareStackTrace callback to avoid passing the JS objects (
-  // the exception and trace) across the realm boundary with the
-  // `Error.prepareStackTrace` override.
   Realm* current_realm = Realm::GetCurrent(context);
+  Local<Function> prepare;
   if (current_realm != nullptr &&
       current_realm->kind() != Realm::Kind::kPrincipal) {
```

Comment:
I think we can just do:

```suggestion
  if (current_realm != nullptr) {
```

here.